client/src/components/Loading.tsx

```diff
@@ -1,10 +1,22 @@
+import { Suspense } from "react";
+import { Await, useLoaderData, Outlet, Navigate } from "react-router-dom";
 import { Box, CircularProgress } from "@chakra-ui/react";
 
 const Loading = () => {
+  const data = useLoaderData() as { response: null };
+
   return (
-    <Box bg="#156087" display="flex" minH="calc(100vh)">
-      <CircularProgress isIndeterminate color="white" m="auto" />
-    </Box>
+    <Suspense
+      fallback={
+        <Box bg="#156087" display="flex" minH="100vh">
+          <CircularProgress isIndeterminate color="orange" m="auto" />
+        </Box>
+      }
+    >
+      <Await resolve={data.response} errorElement={<Navigate to="/login" />}>
```

Comment:
Notice how we are using `errorElement` to navigate the user to `/login` if the implicit login has _any_ error.
Here we consider a user not being logged in as an automatic login error as well, since in that case the response will have an `errorMessage` field.

However, all of these errors are being caught in the `implicitLogin` function, which means the `Await` component will never render the `errorElement`.